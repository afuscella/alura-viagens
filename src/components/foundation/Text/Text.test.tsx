import React from 'react';
import { render, screen } from '../../../infra/tests/test-utils';

import { Text } from './index';

describe('<Text />', () => {
  describe('render component', () => {
    test('with title', () => {
      render(
        <Text variant="title">
          Test
        </Text>,
      );

      const textComponent = screen.getByText('Test');
      expect(textComponent).toMatchSnapshot();
    });

    test('with paragraph', () => {
      render(
        <Text variant="paragraph">
          Test
        </Text>,
      );

      const textComponent = screen.getByText('Test');
      expect(textComponent).toMatchSnapshot();
    });
  });
});
