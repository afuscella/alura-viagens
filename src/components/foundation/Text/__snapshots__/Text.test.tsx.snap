// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> render component with paragraph 1`] = `
.c0 {
  color: #000000;
  font-family: 'Roboto',sans-serif;
  font-size: 18px;
  font-weight: 400;
  line-height: 1;
}

<span
  class="c0"
  color="secondary.main"
>
  Test
</span>
`;

exports[`<Text /> render component with title 1`] = `
.c0 {
  color: #000000;
  font-family: 'Pattaya';
  font-size: 36px;
  font-weight: 400;
  line-height: 1.5;
}

@media only screen and (min-width:768px) {
  .c0 {
    font-family: 'Pattaya';
    font-size: 72px;
    font-weight: 400;
    line-height: 2;
  }
}

<span
  class="c0"
  color="secondary.main"
>
  Test
</span>
`;
